src/simd/avx.cpp

```diff
@@ -155,4 +155,50 @@ PQDistanceAVXFloat256(const void* single_dim_centers, float single_dim_val, void
     }
 }
 
+namespace AVX2 {
+float
+FP32ComputeIP(const float* query, const float* codes, uint64_t dim) {
+#if defined(ENABLE_AVX2)
+    const int n = dim / 8;             // process 8 floats at a time
+    __m256 sum = _mm256_setzero_ps();  // initialize to 0
+    for (int i = 0; i < n; ++i) {
+        __m256 a = _mm256_loadu_ps(query + i * 8);      // load 8 floats from memory
+        __m256 b = _mm256_loadu_ps(codes + i * 8);      // load 8 floats from memory
+        sum = _mm256_add_ps(sum, _mm256_mul_ps(a, b));  // accumulate the product
+    }
+    alignas(32) float result[8];
+    _mm256_store_ps(result, sum);  // store the accumulated result into an array
```

Comment:
When n == 0, the store instruction (e.g. _mm256_store_ps) still occur in the avx512, avx2, and sse operators that may be harmful to the performance. Is it possible to add a judgment to skip these store instructions?

Reply:
yeah, I have fixed it